File: docs/guides/examples/cpp/src/data_ingestion_batch.cc
#include <reduct/client.h>
#include <iostream>
#include <cassert>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;

using std::chrono_literals::operator ""s;

int main() {
    // Create a client instance, then get or create a bucket
    auto client = IClient::Build("http://127.0.0.1:8383", {.api_token="my-token"});
    auto [bucket, create_err] = client->GetOrCreateBucket("my-bucket");
    assert(create_err == Error::kOk);


    // Send a batch of records to the "entry-1" entry
    auto [record_errors, http_err] = bucket->WriteBatch("entry-1", [](auto batch) {
        IBucket::Time ts = IBucket::Time::clock::now();
        batch->AddRecord(ts, "Records #1");
        batch->AddRecord(ts + 1s, "Records #2");
    });
    assert(http_err == Error::kOk);

    // Check if all records were sent successfully
    for (auto& [ts, err] : record_errors) {
        assert(err == Error::kOk);
    }

    return 0;
}

File: docs/guides/examples/py/src/data_ingestion_batch.py
import time
import asyncio
from typing import Dict

from reduct import Client, Bucket, Batch, ReductError


async def main():
    # Create a client instance, then get or create a bucket
    client = Client("http://127.0.0.1:8383", api_token="my-token")
    bucket: Bucket = await client.create_bucket("my-bucket", exist_ok=True)

    # Prepare a batch of records
    batch = Batch()
    batch.add("2024-02-02T10:00:00", b"Records #1", )
    batch.add("2024-02-02T10:00:01", b"Records #2", )
    batch.add("2024-02-02T10:00:02", b"Records #3", )

    # Write the batch to the "entry-1" entry of the bucket
    errors: Dict[int, ReductError] = await bucket.write_batch("entry-1", batch)

    # Check statuses and raise first error
    for timestamp, err in errors.items():
        raise err


loop = asyncio.get_event_loop()
loop.run_until_complete(main())
